src/ngx_http_lua_shdict.c

```diff
@@ -619,9 +630,11 @@ ngx_http_lua_shdict_flush_expired(lua_State *L)
         return luaL_error(L, "expecting 1 or 2 argument(s), but saw %d", n);
     }
 
-    luaL_checktype(L, 1, LUA_TLIGHTUSERDATA);
+    luaL_checktype(L, 1, LUA_TTABLE);
 
-    zone = lua_touserdata(L, 1);
+    lua_rawgeti(L, 1, 1);
+    zone = lua_touserdata(L, -1);
+    lua_pop(L, 1);
```

Comment:
Because this code sequence (the 3 lines above) appears frequently, it makes sense to abstract it away with a static (inline) C function like `ngx_http_lua_shdict_get_zone`. What do you think?